eyeshade/controllers/owners.js

```diff
@@ -154,46 +159,100 @@ const bulk = async (request, reply, runtime, owner, info, providers) => {
   }
   await owners.update({ owner: owner }, state, { upsert: true })
 
-  for (let entry of providers) {
-    const previous = await publishers.findOne({ publisher: entry.publisher })
+  for (let channel of channels) {
+    const previous = await publishers.findOne({ publisher: channel.channelId })
 
     if ((previous) && (previous.owner) && (previous.owner !== owner)) {
       runtime.notify(debug, {
         channel: '#publishers-bot',
-        text: 'new owner ' + ownerString(owner, info) + 'for ' + entry.publisher + ', previously' +
+        text: 'new owner ' + ownerString(owner, info) + 'for ' + channel.channelId + ', previously' +
           ownerString(previous.owner, previous.info)
       })
     }
 
-    state.$set = underscore.extend(underscore.omit(entry, [ 'publisher', 'show_verification_status' ]), {
+    props = batPublisher.getPublisherProps(channel.channelId)
+
+    state.$set = underscore.extend(underscore.omit(channel, [ 'channelId' ]), {
       verified: true,
       authorized: true,
       authority: owner,
       owner: owner,
-      visible: entry.show_verification_status || false,
       altcurrency: altcurrency,
-      info: info
-    })
+      info: underscore.defaults({ name: channel.authorizerName, email: channel.authorizerEmail }, info)
+    }, underscore.pick(props, [ 'providerName', 'providerSuffix', 'providerValue' ]))
 
-    await publishers.update({ publisher: entry.publisher }, state, { upsert: true })
+    await publishers.update({ publisher: channel.channelId }, state, { upsert: true })
 
-    entry.verificationId = uuid.v4().toLowerCase()
+    channel.verificationId = uuid.v4().toLowerCase()
     state.$set = underscore.extend(underscore.pick(state.$set, [ 'verified', 'visible' ]), {
-      token: entry.verificationId,
+      token: channel.verificationId,
       reason: 'bulk loaded',
       authority: owner,
       info: info
     })
-    await tokens.update({ publisher: entry.publisher, verificationId: entry.verificationId }, state, { upsert: true })
+    await tokens.update({ publisher: channel.channelId, verificationId: channel.verificationId }, state, { upsert: true })
 
     await runtime.queue.send(debug, 'publisher-report',
-                             underscore.extend({ owner: owner, publisher: entry.publisher },
+                             underscore.extend({ owner: owner, publisher: channel.channelId },
                                                underscore.pick(state.$set, [ 'verified', 'visible' ])))
   }
 
   reply({})
 }
 
+/*
+   DELETE /v1/owners/{owner}/{publisher}
+       [ used by publishers ]
+ */
+
+v1.deletePublisher = {
```

Comment:
nit: maybe `unlinkPublisher` here to match description?

Reply:
i don't think it really matters, but since i'm going to edit the file anyway... why not?